openedx-unsupported/configuration
# OLIVE-19 Enable automatic provisioning of jenkins instance dedicated to running analytics tasks.

# Testing instructions
1. Create a virtualenv and install `requirements.txt` there. 
2. Verify that you have access to: https://github.com/edx-ops/edx-jenkins-job-dsl/
3. Generate a new ssh-key. 
4. Attach this ssh-key to your github account. 
5. Create a yaml file containing configuration, required configuration is explained in `Readme.md` (if it's not obvious what to put in the file it's a documentation bug :))
6. Go to `vagrant/base/analytic_jenkins` and issue `VAGRANT_JENKINS_LOCAL_VARS_FILE=your/path vagrant up`
7. Expected result is that you will get a working jenkins istance that you'll be able to log in to `192.168.33.15:8080` or `localhost:8080`. 
8. You'll be able to login using `jenkins:password you set` credentials to the instance. 
9. There will be a role called `seed` which role was already ran single time with success result.
10. During the run `seed` role created a new role called: `stub-analytics-jobs`. 
11. Remove the key you added to github in step 2. :) 
# Documentation check
1. Check documentation of the new role `playbooks/roles/jenkins_analytics/README.md` to check whether you understand how to configure the this role. 
2. Check outstanding issues to check which ones are grievous enough to fix right now :)
